test/dm.c

```diff
@@ -58,53 +59,61 @@ static void flip_bits_filename(gchar *filename, off_t offset, guint8 mask)
 	g_close(fd, NULL);
 }
 
-static void drop_caches(void)
+static guint readable_sectors(int fd, GBytes *original)
 {
-	int fd = g_open("/proc/sys/vm/drop_caches", O_WRONLY|O_CLOEXEC, 0);
-	g_assert_cmpint(fd, >, 0);
-	g_assert_true(write(fd, "1", 1) == 1);
-	g_close(fd, NULL);
-}
-
-static guint readable_sectors(int fd)
-{
-	guint8 buf[4096];
+	g_autofree guint8 *buf = NULL;
 	ssize_t r;
 	guint sectors = 0;
 
+	/* O_DIRECT needs paged-aligned memory */
+	buf = aligned_alloc(4096, 4096);
```

Comment:
RAUC configures `AC_PROG_CC_C99`, but `aligned_alloc` is C11. Alternative would be `posix_memalign` (or mandate C11?).

Reply:
Apparently, you need a quite recent autotools to be able to configure C11 out of the box. Given that we want to move to meson, it may not be worth it opencoding something at our side. We don't have an error using `aligned_alloc` because autotools generated config.h has `#define _GNU_SOURCE`, which loosens the requirement that headers only export symbols mandated by the standard as not to clash with user-defined names. `aligned_alloc` has been part of glibc since v2.16 released in 2012, so I guess we can just stay with the implicit assumption that everyone running the test has a recent enough glibc and when moving to Meson, this can become explicit by mandating C11.